### ISSUE TITLE
Two notations with the same symbol used simultaneously

### ISSUE THREAD

**Issue by eponier** [coq/coq#12908]
#### Description of the problem

I tried to define my own `mult` function, and use `*` as a notation for it. But then, `*` is used both for my own `mult` function, and for the standard `mul` function defined in the standard library, which can give quite unreadable statements.

For instance,

```
Definition mult' m n := 2 * m * n.

Infix "*" := mult'. (* with or without ": nat_scope" *)

Goal forall m n, mult' m n = Nat.mul (Nat.mul 2 m) n.
Proof.
  (* forall m n, m * n = 2 * m * n *)
  (* the "*" on the left is [mult'], the "*" on the right is [Nat.mul] *)

```
#### Coq Version

8.12

**Comment by eponier** [comment]
Sorry if an issue already existed for this problem, but there are so many issues related to the notation system that it is quite hard to search for a specific one.

**Comment by herbelin** [comment]
That's somehow curious that this basic undetected conflict between notations remained unnoticed for so long. But better late than never and thanks for reporting.

In the case without a scope, this is fixed by #12946. For the case with a scope, this is fixed in #12950.

**Comment by herbelin** [comment]
Fixed in #12946 and #12950.
